### PR TITLE
Feature/healthchecks plugin

### PR DESCRIPTION
**Pull request by matilote**

### CODE REVIEW

File: src/Nethermind/Nethermind.HealthChecks/NodeHealthChecks.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+                if(ethSyncing.IsSyncing == false && netPeerCount > 0)
+                {
+                    return HealthCheckResult.Healthy(description: $"The node is now fully synced with a network, number of peers: {netPeerCount}");
+                }
+                else if (ethSyncing.IsSyncing == false && netPeerCount == 0)
+                {
+                    return HealthCheckResult.Unhealthy(description: $"The node has 0 peers connected");
+                }
+                else
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.HealthChecks/NodeHealthChecks.cs:52]: if you want you can delete this else statement and simply return after previous if and else if was false
just a minor tip, can be ignored

File: src/Nethermind/Nethermind.HealthChecks/HealthChecksConfig.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+namespace Nethermind.HealthChecks
+{
+    public class HealthChecksConfig : IHealthChecksConfig
+    {
+        public bool Enabled { get; set; } = false;
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.HealthChecks/HealthChecksConfig.cs:22]: Documentation missing

File: src/Nethermind/Nethermind.HealthChecks/IHealthChecksConfig.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        public string WebhooksPayload { get; set; }
+
+        [ConfigItem(Description = "RestorePayload is the json payload that will be send on Recovery and must be escaped.", DefaultValue = "{\"attachments\":[{\"color\":\"#36a64f\",\"pretext\":\"Health Check Status :+1:\",\"fields\":[{\"title\":\"Details\",\"value\":\"`More details available at /healthchecks-ui`\",\"short\":false},{\"title\":\"description\",\"value\":\"The HealthCheck `[[LIVENESS]]` is recovered. All is up and running\",\"short\":false}]}]}")]
+        public string WebhooksRestorePayload { get; set; }
+
+        [ConfigItem(Description = "If 'true' then HealthChecks UI will be avaiable at /healthchecks-ui", DefaultValue = "false")]
+        public bool UIEnabled { get; set; }
+
+        [ConfigItem(Description = "Configures the UI to poll for healthchecks updates", DefaultValue = "5")]
+        public int PollingInterval { get; set; }
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.HealthChecks/IHealthChecksConfig.cs:43]: Always tell user whether it is miliseconds or seconds.

File: src/Nethermind/Nethermind.HealthChecks/IHealthChecksConfig.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public interface IHealthChecksConfig : IConfig
+    {
+        [ConfigItem(Description = "If 'true' then Health Check endpoints is enabled at /health", DefaultValue = "false")]
+        public bool Enabled { get; set; }
+
+        [ConfigItem(Description = "If 'true' then Webhooks can be configured", DefaultValue = "false")]
+        public bool WebhooksEnabled { get; set; }
+
+        [ConfigItem(Description = "The Webhooks endpoints e.g. Slack WebHooks", DefaultValue = "null")]
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.HealthChecks/IHealthChecksConfig.cs:30]: why 'endpoints' if only one allowed

File: src/Nethermind/Nethermind.Runner/Ethereum/JsonRpcRunner.cs [hunk 5]
```diff
@@ -45,12 +46,14 @@ public class JsonRpcRunner
...
         private IInitConfig _initConfig;
+        private INethermindApi _api;
 
         public JsonRpcRunner(
             IJsonRpcProcessor jsonRpcProcessor,
             IWebSocketsManager webSocketsManager,
             IConfigProvider configurationProvider,
-            ILogManager logManager)
+            ILogManager logManager,
+            INethermindApi api)
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/Ethereum/JsonRpcRunner.cs:56]: Is API initialized at this point?

Comment (tkstanczak) [reply src/Nethermind/Nethermind.Runner/Ethereum/JsonRpcRunner.cs:56]: why do you need api here?

File: src/Nethermind/Nethermind.Runner/Ethereum/JsonRpcRunner.cs [hunk 6]
```diff
@@ -102,6 +106,10 @@ IEnumerable<string> GetUrls()
                     s.AddSingleton(_configurationProvider);
                     s.AddSingleton(_jsonRpcProcessor);
                     s.AddSingleton(_webSocketsManager);
+                    foreach(var plugin in _api.Plugins.OfType<INethermindServicesPlugin>()) 
+                    {
+                        plugin.AddServices(s);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/Ethereum/JsonRpcRunner.cs:111]: I see

File: src/Nethermind/Nethermind.HealthChecks/NodeHealthChecks.cs (hunk 1 above)
Comment (tkstanczak) [reply src/Nethermind/Nethermind.HealthChecks/NodeHealthChecks.cs:52]: What I started doing in all my new code is to have only one return statement per method. I was avoiding this for a long time but now I have to admit it helps a lot.

**Review by tkstanczak — APPROVED** [review]

**Pull request merged**